src/main/java/org/folio/rest/impl/EHoldingsRootProxyImpl.java

```diff
@@ -30,35 +28,27 @@
 import org.folio.rest.validator.RootProxyPutBodyValidator;
 import org.folio.rmapi.RMAPIService;
 import org.folio.rmapi.exception.RMAPIUnAuthorizedException;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
 
 public class EHoldingsRootProxyImpl implements EholdingsRootProxy {
 
+  @Autowired
   private RMAPIConfigurationService configurationService;
+  @Autowired
   private HeaderValidator headerValidator;
+  @Autowired
   private RootProxyConverter converter;
+  @Autowired
   private RootProxyPutBodyValidator bodyValidator;
 
   private static final String PUT_ROOT_PROXY_ERROR_MESSAGE = "Failed to update root proxy";
 
   private final Logger logger = LoggerFactory.getLogger(EHoldingsRootProxyImpl.class);
 
-  public EHoldingsRootProxyImpl() {
-    this(
-      new RMAPIConfigurationServiceCache(
-        new RMAPIConfigurationServiceImpl(new ConfigurationClientProvider())),
-      new HeaderValidator(),
-      new RootProxyConverter(),
-      new RootProxyPutBodyValidator());
-  }
-
-  public EHoldingsRootProxyImpl(RMAPIConfigurationService configurationService,
-                                HeaderValidator headerValidator,
-                                RootProxyConverter converter,
-                                RootProxyPutBodyValidator bodyValidator) {
-    this.configurationService = configurationService;
-    this.headerValidator = headerValidator;
-    this.converter = converter;
-    this.bodyValidator = bodyValidator;
+  @SuppressWarnings("squid:S1172")
```

Comment:
Will this be required now for all constructors due to di?

Reply:
Yes, this is not related to di, Sonar generates a warning here because tenanId isn't used, but in reality RestVerticle uses reflection to find a constructor with Vertx and String parameters. So without tenantId parameter RestVerticle won't call this constructor.
Here's code from RestVerticle.
```
try {
    o = aClass.getConstructor(Vertx.class, String.class).newInstance(this.vertx, tenantId[0]);
} catch (Exception var36) {
    o = aClass.newInstance();
}
```

Reply:
A little research of Vertx code has shown that we can use default constructor and obtain Vertx instance from the current context:
- `VertxUtils.getVertxFromContextOrNew()` or
- 'Vertx.currentContext().owner()'

2nd case is more strict and states that we don't create a new Vertx instance since it should be already available